scan.go

```diff
@@ -671,6 +689,27 @@ func (zp *ZoneParser) Next() (RR, bool) {
 	return nil, false
 }
 
+// canParseAsRR returns true if the record type can be parsed as a
+// concrete RR. It blacklists certain record types that must be parsed
+// according to RFC 3597 because they lack a presentation format.
+func canParseAsRR(rrtype uint16) bool {
+	switch rrtype {
+	case TypeANY, TypeNULL, TypeOPT, TypeTSIG:
+		return false
+	default:
+		return true
+	}
+}
+
+func allowDynamicUpdate(rr RR) bool {
+	switch rr.(type) {
+	case *TKEY, *RFC3597, *PrivateRR:
```

Comment:
Think we should only check for PrivateRRs

Reply:
Acknowledged.